### PR TITLE
Print routes lexicographically and fix `$routes` resolution bug. 

### LINKED ISSUE #703: Print method does not print trailing slashes in endpoints
Trailing slashes are not respected when printing

Reprex:

```
r <- pr() %>% pr_get("/A/B/", identity)
r
#> # Plumber router with 1 endpoint, 4 filters, and 0 sub-routers.
#> # Call run() on this object to start the API.
#> ├──[queryString]
#> ├──[body]
#> ├──[cookieParser]
#> ├──[sharedSecret]
#> ├──/A
#> │  └──/B (GET)
```

I'd expect:

```
#> # Plumber router with 1 endpoint, 4 filters, and 0 sub-routers.
#> # Call run() on this object to start the API.
#> ├──[queryString]
#> ├──[body]
#> ├──[cookieParser]
#> ├──[sharedSecret]
#> ├──/A
#> │  └──/B 
#> │     └──/ (GET)
```

### PR DESCRIPTION
**Pull request by meztez**
Reprex in #701 
Closes #703 

PR task list:
- [x] Update NEWS
- [x] Add tests
- [ ] Update documentation with `devtools::document()`

### CODE REVIEW

File: R/plumber.R [hunk 1]
```diff
@@ -1020,7 +1028,20 @@ Plumber <- R6Class(
...
+            logical(1),
+            "PlumberEndpoint")
+        if (any(existing_endpoints) && length(children) > 1) {
+          node <- c(
+            node[which(names(node) == children[1])][which(existing_endpoints)],
+            addPath(node[which(names(node) == children[1])][which(!existing_endpoints)], children, endpoint)
+          )
+        } else {
+          node[[children[1]]] <- addPath(node[[children[1]]], children[-1], endpoint)
+        }
```
Comment (schloerke) [inline R/plumber.R:1044]: These are pretty complex statements. 

Do you mind adding a few comments / reasons for the code paths?

Thank you!

File: R/plumber.R [hunk 2]
```diff
@@ -443,7 +443,15 @@ Plumber <- R6Class(
...
             } else {
               # there are other endpoints, so get only nodes with name ""
-              names(node) == ""
+              # which path does not end with / and path is not root
+              node_path <- function(node) {
+                if (!is.null(node$path)) {
+                  return(node$path)
+                } else {
+                  return(character(1))
+                }
```
Comment (schloerke) [inline R/plumber.R:452]:
```suggestion
                node$path %||% ""
```

File: R/plumber.R [hunk 3]
```diff
@@ -1030,6 +1051,7 @@ Plumber <- R6Class(
           path <- sub("^/", "", e$path)
 
           levels <- strsplit(path, "/", fixed=TRUE)[[1]]
+          if (grepl("/$", path)) {levels <- c(levels, "")}
```
Comment (schloerke) [inline R/plumber.R:1054]:
```suggestion
          # If there is a trailing `/`, add a blank level for an extra print line
          if (grepl("/$", path)) {levels <- c(levels, "")}
```

**Review by schloerke — CHANGES_REQUESTED** [review]

File: R/plumber.R (hunk 2 above)
Comment (meztez) [reply R/plumber.R:452]: what is this black magic operator, I want in.

File: R/plumber.R (hunk 1 above)
Comment (meztez) [reply R/plumber.R:1044]: Sure

**Comment by schloerke** [comment]
Closing in favor of https://github.com/rstudio/plumber/pull/737

**Pull request closed without merging**